docs/source/kedro_datasets.rst

```diff
@@ -36,12 +36,16 @@ kedro_datasets
    kedro_datasets.pandas.SQLQueryDataset
    kedro_datasets.pandas.SQLTableDataset
    kedro_datasets.pandas.XMLDataset
+   kedro_datasets.partitions.IncrementalDataset
+   kedro_datasets.partitions.PartitionedDataset
    kedro_datasets.pickle.PickleDataset
    kedro_datasets.pillow.ImageDataset
    kedro_datasets.plotly.JSONDataset
    kedro_datasets.plotly.PlotlyDataset
    kedro_datasets.polars.CSVDataset
    kedro_datasets.polars.GenericDataset
+   kedro_datasets.polars.EagerPolarsDataset
```

Comment:
These datasets do not exist in `kedro-datasets[all]~=1.7.0`, the version currently specified in `pyproject.toml`.